frontend/src/components/Sidebar/HeadlampButton.tsx

```diff
@@ -56,11 +61,24 @@ export default function HeadlampButton({ open, onToggleOpen, mobileOnly }: Headl
         className={classes.button}
         aria-label={open ? t('Shrink sidebar') : t('Expand sidebar')}
       >
-        <SvgIcon
-          className={classes.logo}
-          component={open ? LogoWithTextLight : LogoLight}
-          viewBox="0 0 auto 32"
-        />
+        {
+          // Till all plugins are not loaded show empty content for logo as we might have logo coming from a plugin
+          pluginLoadState === false ? (
+            <EmptyContent />
+          ) : AppLogoGeneratorComponent ? (
+            <AppLogoGeneratorComponent
+              logoType={open ? LOGO_TYPE.LARGE : LOGO_TYPE.SMALL}
```

Comment:
You could get rid of LOGO_TYPE here if you used `"small" : "large"` and relied on typescript to check the strings.